### PR TITLE
feat(css_formatter): Bootstrap the CSS Formatter!

### PR DESCRIPTION
**Pull request by faultyserver**
## Summary

Styles!

This PR bootstraps all of the infrastructure for `biome_css_formatter`, including:
- the `biome_css_formatter` crate itself
- configuration through `biome_cli`
- configuration through `biome_service`
- adding `CssFileSource` to be able to load css files as well as text from stdin
- adding some codegen logic to handle css things specifically

Also included are a few other things that stuck out while I was going through all of the files:
- Moved `JsonParserSettings` out of `file_handlers/javascript.rs` and into `file_handlers/json.rs`.
- Fixed some typos in various places that were relevant
- Updated READMEs and comments to reference the appropriate crate (i.e., "run `cargo test biome_json_formatter`", but it referenced `biome_js_formatter` before).

This doesn't do any actual formatting yet! It's just the framework so that we can start writing tests and implementing the formatting itself in the future.

Going through this whole process, I definitely think we need to reduce the amount of boilerplate needed to add a new language. Things like configuration options have to get duplicated in like 7 places currently, there's a lot of duplicated code across languages that could possibly be done with generics? Maybe some other things as well. I'm not totally sure, but I'm positive that it's not a great experience if we ever want to open up a plugin system. But that's something to solve another day!

## Test Plan

All the tests are passing and utilize CSS configuration options and such to make sure all of that is wired up correctly, plus the basic spec and smoke test for `biome_css_formatter` passes.

### CODE REVIEW

**Comment by ematipico** [comment]
Thank you @faultyserver , that's exciting!! 

How do you plan to release this to the users? If we decided to merge the PR as is, users would think that the formatter would be ready for format CSS files, instead it's not like this.

My suggestion would be to **hide** the CLI options, via the `hide` macro option, and keep CSS files still **unhandled** by the workspace.

File: crates/biome_css_formatter/LICENSE [hunk 1]
```diff
@@ -0,0 +1,25 @@
+MIT License
```
Comment (ematipico) [inline crates/biome_css_formatter/LICENSE:1]: This is the Rome Tools license, we don't need it. We can delete this file

**Review by ematipico — COMMENTED** [review]
Awesome work!!

> Things like configuration options have to get duplicated in like 7 places currently, there's a lot of duplicated code across languages that could possibly be done with generics? Maybe some other things as well. I'm not totally sure, but I'm positive that it's not a great experience if we ever want to open up a plugin system. But that's something to solve another day!

The best way to reduce boilerplate in this case is by using macros

**Comment by faultyserver** [comment]
> How do you plan to release this to the users? If we decided to merge the PR as is, users would think that the formatter would be ready for format CSS files, instead it's not like this.
> My suggestion would be to hide the CLI options, via the hide macro option, and keep CSS files still unhandled by the workspace.

That sounds like a good plan to me. I think hiding the CLI options is pretty straightforward, but I'm not sure about how to wire up the workspace for that. I could just remove the `capabilities` maybe? That way the configuration is still valid to add and use, it just won't actually do anything when it runs. It might be strange for users to be able to configure something that doesn't currently "exist" from their perspective, but I figure that's probably fine to allow, since most people won't even notice that it _can_ be added yet.

**Comment by ematipico** [comment]
That's actually pretty simple, when retrieving the language for the file extension, we can just return `Unknown` and the logic should just work like the CSS files are still unrecognised

**Comment by faultyserver** [comment]
Okay, with the hiding in place, running `format` on a css file continues to show that it is "unhandled", which i think is what we want:
image

**Comment by ematipico** [comment]
As a cherry on top, we should update the website, language support page, and put the hourglass emoji on the format cell

**Review by ematipico — APPROVED** [review]

**Pull request merged**